client/widgets/markets/TradePanels.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * TradePanels.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#pragma once
+
+#include "../Images.h"
+
+#include "../../../lib/networkPacks/TradeItem.h"
+
+class CTextBox;
+class SelectableSlot;
+
+enum class EType
+{
+	RESOURCE, PLAYER, ARTIFACT_TYPE, CREATURE, CREATURE_PLACEHOLDER, ARTIFACT_PLACEHOLDER, ARTIFACT_INSTANCE
+};
+
+class CTradeableItem : public CIntObject, public std::enable_shared_from_this<CTradeableItem>
+{
+public:
+	std::shared_ptr<CAnimImage> image;
+	AnimationPath getFilename();
+	int getIndex();
+	using ClickPressedFunctor = std::function<void(const std::shared_ptr<CTradeableItem>&)>;
+
+	const CArtifactInstance * hlp; //holds ptr to artifact instance id type artifact
```

Comment:
use better name?
```suggestion
	const CArtifactInstance * artInstance; //holds ptr to artifact instance id type artifact
```

Reply:
> use better name?

CTradeableItem will be reworked. It's just moving the code. I also ignored warnings related to it in sonarcloud